src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -327,23 +347,32 @@ def _process_mash_tool(root_dir: str,
                 metadata = json.load(file)
 
             sketch_file = metadata['sketch_file']
+            sketch_data = _read_sketch(sketch_file)
+            sketches = sketch_data['sketches']
+            if len(sketches) != 1:
+                raise ValueError(f'Expected only one sketch in the mash info output for genome: {data_id}')
+            sketch_id = sketches[0]['name']
+            seq_meta.append({'sourceid': data_id, 'id': sketch_id})
             if not os.path.exists(sketch_file):
                 raise ValueError(f'Unable to locate the sketch file: {sketch_file} for genome: {data_id}')
             sketch_files.append(sketch_file)
 
+    _create_import_files(root_dir, f'{kbase_collection}_{load_ver}_{SEQ_METADATA}', seq_meta)
+
     import_dir = Path(root_dir, IMPORT_DIR)
     os.makedirs(import_dir, exist_ok=True)
     mash_output_prefix = import_dir / f'{kbase_collection}_{load_ver}_merged_sketch'
 
     # write the lines from sketch_files into a temporary file
-    with tempfile.NamedTemporaryFile(mode='w', delete=True) as temp_file:
+    with tempfile.NamedTemporaryFile(mode='w', delete=False) as temp_file:
         temp_file.write('\n'.join(sketch_files))
         temp_file_path = temp_file.name
 
-        # run mash paste
-        command = ['mash', 'paste', mash_output_prefix, '-l', temp_file_path]
-        print(f'Running mash paste: {command}')
-        run_command(command)
+    # run mash paste
+    command = ['mash', 'paste', str(mash_output_prefix), '-l', temp_file_path]
+    print(f'Running mash paste: {" ".join(command)}')
+    run_command(command)
+    os.remove(temp_file_path)
```

Comment:
I'm confused here - the temp file should exist while mash is running with the old implementation, right? The original implementation looks right

Reply:
I think before the file is closed, the file content is empty. Thus mash paste creates empty msh file.

Reply:
But why is the file empty?

Reply:
`In [5]:         sketch_files = ['a', 'b', 'c']
   ...:         with tempfile.NamedTemporaryFile(mode='w', delete=False) as temp_file:
   ...:             temp_file.write('\n'.join(sketch_files))
   ...:             temp_file_path = temp_file.name
   ...: 
   ...:             with open(temp_file_path, 'r') as file:
   ...:                 content = file.read()
   ...:                 print(f'file content: {content}')
   ...: 
file content: 

`

Reply:
![Screenshot 2023-06-06 at 8 29 30 AM](https://github.com/kbase/collections/assets/3953823/ddc0c12d-69db-45e9-a65e-425e99770ab2)

Reply:
Maybe add `temp_file.flush()`?

Reply:
👍

Reply:
Surprised that's necessary but at least it works and is a lot simpler than the other solutions

Reply:
:+1: